### PR TITLE
ci: add markdown linter

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Adds a Markdown Linter - needs some testing before merge - I'll review the `hold` label and add reviewers once I feel it's ready

### CODE REVIEW

File:  .markdownlint-cli2.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  no-duplicate-header: false
+  single-trailing-newline: false
+globs:
+  - "**/*.md"
+ignores:
+  - "pkg"
+  - "**/node_modules/*"
+  - "**/node_modules/**"
+  - ".tox/**"
+  - "venv/**"
```
Comment (tiran) [inline  .markdownlint-cli2.yaml:15]: We don't need node stuff. It's common to use either `.venv` or `venv` for a local development virtual env.
```suggestion
  - ".tox/**"
  - "venv/**"
  - ".venv/**"
```

File: Makefile [hunk 2]
```diff
@@ -127,3 +127,24 @@ verify: ## Run tox -e fmt,lint,spellcheck against code
...
+else
+    ECHO_PREFIX=@\#
+    CMD_PREFIX=
+    PIPE_DEV_NULL=
+endif
+
+.PHONY: md-lint
+md-lint: ## Lint markdown files
+	$(ECHO_PREFIX) printf "  %-12s ./...\n" "[MD LINT]"
+	$(CMD_PREFIX) podman run --rm -v $(CURDIR):/workdir --security-opt label=disable docker.io/davidanson/markdownlint-cli2:v0.6.0 > /dev/null
```
Comment (tiran) [inline Makefile:150]: userns keep-id is better than disabling SELinux:

```suggestion
	$(CMD_PREFIX) podman run --rm -v $(CURDIR):/workdir:ro :ro --userns=keep-id docker.io/davidanson/markdownlint-cli2:v0.6.0 > /dev/null
```

**Review by tiran — COMMENTED** [review]
I like the idea to link markdown files, too.

File: Makefile (hunk 2 above)
Comment (nathan-weinberg) [reply Makefile:150]: I get the following error when trying this on Fedora
```bash
[nathan@nathan-redhat instructlab (md-lint)]$ make md-lint 
  [MD LINT]    ./...
Error: invalid reference format
make: *** [Makefile:150: md-lint] Error 125
```

Comment (russellb) [reply Makefile:150]: same error on mac fwiw

**Comment by nathan-weinberg** [comment]
This should be good to go

File: notebooks/README.md [hunk 3]
```diff
@@ -48,9 +49,7 @@ and the NVIDIA P100 provided in the free tier, we will fine tune a LoRA.
 1. You'll need a Kaggle account, which you can create by visiting [Kaggle's Sign-up Page](https://www.kaggle.com/account/login?phase=startRegisterTab&returnUrl=%2F).
 1. To use Kaggle's accelerators, you'll have to verify your account with a phone number. Visit the [account settings page](https://www.kaggle.com/settings) and select "Phone Verification".
 
-
-**NOTE: At present, you'll need to download the notebook and upload it to Kaggle. The following steps will walk you through uploading the notebook. Once this repository is open sourced, we will make an 'Open in Colab' button**
-
+##### NOTE: At present, you'll need to download the notebook and upload it to Kaggle. The following steps will walk you through uploading the notebook. Once this repository is open sourced, we will make an 'Open in Colab' button**
```
Comment (russellb) [inline notebooks/README.md:52]: It can be another PR, but this formatting is nicer:

https://github.com/orgs/community/discussions/16925

File: src/instructlab/generator/README.md [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ### Notice
-This folder uses code from the `stanford_alpaca` project developed by Tatsu Lab, available at https://github.com/tatsu-lab/stanford_alpaca. The original code is licensed under the Apache License:
 
-```
+This folder uses code from the `stanford_alpaca` project developed by Tatsu Lab, available [here](https://github.com/tatsu-lab/stanford_alpaca). The original code is licensed under the Apache License:
+
+```license
```
Comment (russellb) [inline src/instructlab/generator/README.md:5]: Do you know for sure this is valid?  I don't think it'll hurt if it's not recognized. I would have just done `text` probably. I won't block on it since I don't think it breaks anything

Comment (nathan-weinberg) [reply src/instructlab/generator/README.md:5]: I see it being used other places in the repo, so I assume it's valid

File: notebooks/README.md (hunk 3 above)
Comment (nathan-weinberg) [reply notebooks/README.md:52]: simple enough to change, will do that as part of this PR

**Review by russellb — APPROVED** [review]
One follow-up is to block merging on this job passing in the mergify config

File: Makefile (hunk 2 above)
Comment (nathan-weinberg) [reply Makefile]: We can address this in a followup

**Comment by russellb** [comment]
@mergifyio refresh
